beetsplug/fetchart.py

```diff
@@ -308,16 +308,42 @@ class CoverArtArchive(RemoteArtSource):
     VALID_THUMBNAIL_SIZES = [250, 500, 1200]
 
     if util.SNI_SUPPORTED:
-        URL = 'https://coverartarchive.org/release/{mbid}/front'
-        GROUP_URL = 'https://coverartarchive.org/release-group/{mbid}/front'
+        URL = 'https://coverartarchive.org/release/{mbid}'
+        GROUP_URL = 'https://coverartarchive.org/release-group/{mbid}'
     else:
-        URL = 'http://coverartarchive.org/release/{mbid}/front'
-        GROUP_URL = 'http://coverartarchive.org/release-group/{mbid}/front'
+        URL = 'http://coverartarchive.org/release/{mbid}'
+        GROUP_URL = 'http://coverartarchive.org/release-group/{mbid}'
 
     def get(self, album, plugin, paths):
         """Return the Cover Art Archive and Cover Art Archive release group URLs
         using album MusicBrainz release ID and release group ID.
         """
+
+        def get_image_url(url, size_suffix=None):
+            try:
+                response = self.request(url)
+            except requests.RequestException:
+                self._log.debug(u'{0}: error receiving response'
+                                .format(self.NAME))
+                return
+
+            try:
+                data = response.json()
+            except ValueError:
+                self._log.debug(u'{0}: error loading response: {1}'
+                                .format(self.NAME, response.text))
+                return
+
+            if 'images' in data.keys():
+                for item in data['images']:
+                    if not item['front']:
+                        continue
+
+                    if size_suffix:
+                        return item['thumbnails'][size_suffix]
+
+                    return item['image']
```

Comment:
> Instead of blindly selecting the first image, we now treat all "front" images as candidates.

Since you return instead of `yield`ing here, I think the original intent of the PR is not achieved, is it? This would of course require a `yield from` (or rather a loop) below.